build-showmewebcam.sh

```diff
@@ -26,11 +30,13 @@ esac
 CONFIG_="BR2" KCONFIG_CONFIG="configs/${BOARDNAME}_defconfig" "$BUILDROOT_DIR/support/kconfig/merge_config.sh" -m -r configs/config "configs/$BOARDNAME"
 sed "1i ### DO NOT EDIT, this file was automatically generated\n" -i "configs/${BOARDNAME}_defconfig"
 
-# Merge kernel configurations
-if [ -f "board/linux-${BOARDNAME}.config" ]; then
```

Comment:
Overlays would be possible with this. The better approach would be to make linux-base.config usable for the Pi4 as well.

Reply:
Like I mentioned in my comment, I don't think it's feasible to use the same base config for the pi zero and pi 4. They use different generations of SoC, the pi zero is Armv6 and the Pi 4 is v8 (although here it's only running as armv7l). Making an overlay for the base config currently in master would be disabling a lot of SoC specific options and enabling others which would be a lot more likely to hit issues down the line, as well as a lot more confusing to maintain.

Flipping that idea on it's head, using the defconfig for the board, and a common overlay to disable useless modules and builtins means that the same optimisations don't need to be done by hand for every board.

Reply:
As I said before, defconfig is not the way to go, as we'll loose all optimizations. Flipping this on its head won't work either, as we'll always chasing new and changed options when upgrading the kernel. Hence the better way, still, is to have a base linux config and add all further changes to their variants. If that means moving SoC specific options to their corresponding files, I won't have any problems with that.

Reply:
Is there anything that details what optimisations have been made already to the rpi0?

Would you envisage this being a common base against other board targets too, for example when using a Rockchip RK3399? or a Qualcomm target?

Going to AARCH64 may also have quite an impact on the kernel configurations, and of course all of this has to be maintainable as the kernel version itself is bumped to newer releases?

Reply:
> Is there anything that details what optimisations have been made already to the rpi0?

Kind of here: https://github.com/showmewebcam/showmewebcam/commit/b680695687099045befdce3174a4c6db4f7166d2 and https://github.com/showmewebcam/showmewebcam/commit/a7e933e20ff4e52a965b77616ab64c1b36448e27

> Would you envisage this being a common base against other board targets too, for example when using a Rockchip RK3399? or a Qualcomm target?

I don't see why not

> Going to AARCH64 may also have quite an impact on the kernel configurations, and of course all of this has to be maintainable as the kernel version itself is bumped to newer releases?

Exactly. Hence the approach of using config stubs and merging them with the default buildroot kernel settings.

Reply:
OK, latest version should follow this idea pretty well. Resolving this.